src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -265,4 +267,15 @@ private function getReflectionClass($id, Definition $definition)
             // return null
         }
     }
+
+    private function addServiceToAmbiguousType($id, $type)
+    {
+        // keep an array of all services matching this type
+        if (!isset($this->ambiguousServiceTypes[$type])) {
+            $this->ambiguousServiceTypes[$type] = array(
+                $this->types[$type],
```

Comment:
Are we sure that there always is an entry in `$this->types` for `$type`? In `set()`, we later on explicitly do check for `isset($this->types[$type])`.